Framework/PythonInterface/plugins/algorithms/Abins2D.py

```diff
@@ -0,0 +1,291 @@
+        # so insert placeholder "1" for now.
+        prog_reporter.resetNumSteps(1, 0.1, 0.8)
+
+        s_calculator = abins.SCalculatorFactory.init(filename=self._vibrational_or_phonon_data_file,
+                                                     temperature=self._temperature,
+                                                     sample_form="Powder",
+                                                     abins_data=ab_initio_data,
+                                                     autoconvolution=self._autoconvolution,
+                                                     instrument=self._instrument,
+                                                     quantum_order_num=self._num_quantum_order_events)
+        s_calculator.progress_reporter = prog_reporter
+        s_data = s_calculator.get_formatted_data()
+        self._q_bins = s_data.get_q_bins()
+
+        # Hold reporter at 80% for this message
+        prog_reporter.resetNumSteps(1, 0.8, 0.80000001)
+        prog_reporter.report("Dynamical structure factors have been determined.")
+        # Now determine number of remaining messages and set reporter for rest of run:
+        n_messages = 3 + bool(self._sum_contributions) + bool(self._save_ascii)
+        prog_reporter.resetNumSteps(n_messages, 0.8, 1)
+
+        # 4) get atoms for which S should be plotted
+        atoms_data = ab_initio_data.get_atoms_data()
+        atom_numbers, atom_symbols = self.get_atom_selection(atoms_data=atoms_data, selection=self._atoms)
+        prog_reporter.report("Atoms, for which dynamical structure factors should be plotted, have been determined.")
+
+        # 5) create workspaces for atoms in interest
+        self.setProperty('OutputWorkspace', self._out_ws_name)
+
+        workspaces = []
+        workspaces.extend(self.create_workspaces(atoms_symbols=atom_symbols, s_data=s_data, atoms_data=atoms_data,
+                                                 max_quantum_order=self._max_event_order))
+        workspaces.extend(self.create_workspaces(atom_numbers=atom_numbers, s_data=s_data, atoms_data=atoms_data,
+                                                 max_quantum_order=self._max_event_order))
+        prog_reporter.report("Workspaces with partial dynamical structure factors have been constructed.")
+
+        # 6) Create a workspace with sum of all atoms if required
+        if self._sum_contributions:
+            self.create_total_workspace(workspaces)
+            prog_reporter.report("Workspace with total S has been constructed.")
+
+        GroupWorkspaces(InputWorkspaces=workspaces, OutputWorkspace=self._out_ws_name)
+
+        # 8) save workspaces to ascii_file
+        if self._save_ascii:
+            self.write_workspaces_to_ascii(ws_name=self._out_ws_name,
+                                           scale=(1.0 / self._instrument.get_energy_bin_width()))
+            prog_reporter.report("All workspaces have been saved to ASCII files.")
```

Comment:
is it worth reporting the path to the files?

Reply:
That's not existing behaviour but does seem helpful. Maybe a new Issue?

Reply:
Issue created https://github.com/mantidproject/mantid/issues/33993